src/blog/core2.clj

```diff
@@ -0,0 +1,63 @@
+(ns blog.core
+  (:require [clojure.java.io :as io]
+            [markdown.core :as md]
+            [clojure.string :as str]
+            [clojure.instant :as inst]))
+
+;; Set the directory for the blog post markdown files
+(def blog-dir "./blog/")
+
+;; Let's define the HTML that goes above the blog posts html
+(def html-header
+  (slurp (str blog-dir "static/header.html")))
+
+;; Let's define the HTML that goes below the blog posts html
+(def html-footer
+  (slurp (str blog-dir "static/footer.html")))
+
+; Let's attempt to load the directory for the posts directory
+#_(map #(.toString %) (.list (io/file posts-dir)))
+
+(defn get-md-files
+  "Return a sequence of all markdown files in dir"
+  [dir]
+  (->> (io/file dir)
+       (file-seq)
+       (map #(.toString %))
+       (filter #(str/ends-with? % ".md"))))
```

Comment:
As discussed, this could just return the files. You can use `(.getName f)` in the filter.